packages/block-library/src/utils/hooks.js

```diff
@@ -48,10 +48,7 @@ export function useUploadMediaFromBlobURL( args = {} ) {
 			return;
 		}
 
-		if (
-			! latestArgs.current.url ||
-			! isBlobURL( latestArgs.current.url )
```

Comment:
Let's restore the `isBlobURL` check. I think it's a good safeguard.

Reply:
I'm not sure. If you set "temporaryUrl" you want to explicitly trigger this hook otherwise, why did you use temporaryUrl in the first place?

Reply:
Worse things have happened :D 

The hook expects `args.url` to be a `blob` value. I think an extra check makes sense.

Reply:
This is actually a shared hook so you're right about keeping the check. I restored it.